frontend/apps/filemanager/filemanager.lua

```diff
@@ -363,16 +363,31 @@ function FileManager:init()
 
     if Device:hasKeys() then
         self.key_events.Home = { {"Home"}, doc = "go home" }
-        if not Device:isSDL() then
-            --if not in the desktop emulator
-            --remove the old Back key to exit koreader
-            self.file_chooser.key_events.Close = nil
-        end
+        --Override the menu.lua way of handling the back key
+        self.file_chooser.key_events.Back = { {"Back"}, doc = "go back" }
     end
 
     self:handleEvent(Event:new("SetDimensions", self.dimen))
 end
 
+function FileChooser:onBack()
+    local back_to_exit = G_reader_settings:readSetting("back_to_exit") or "prompt"
```

Comment:
So I was thinking the logic here might not necessarily default to the same thing on all platforms. Particularly on Android.

Also since this is duplicated it might make more sense to include it from a file in `frontend/ui/elements`, although we should keep in mind that from the reader we probably want to exit to filemanager first. I suppose it might be better to work out the quirks first before refactoring.

Reply:
I vehemently dislike quitting or even asking to quit in this manner. I have a million ways to quit, none of which involve accidentally quitting just because I was backing out of a dialog.

Reply:
If we are talking only about the physical back button on Android phones, which is working for leaving a dialog, it should end there.  If invoked outside of a dialog, it could return to the calling Android menu page/exit KOReader.  (On my phone, that would still leave it on the recent apps page where I can select it and it opens on the page it left from.)

Reply:
On Android it's fine. I might dislike it as a general principle but it's the Android way. The other reading app that makes back to exit a "press back again to exit" affair implements a pretty clever compromise.

I'm talking mainly from the perspective of using it as a desktop app, I suppose, particularly in combination with a gamepad. I've actually been doing that recently and I've been entertaining the idea of making a Snap out of KOReader. Or in other words, doing #1417. The only big missing feature compared to Zathura, the main competitor the way I see it, is dual-page view and a couple of keyboard shortcuts. It might sound slightly odd, but it's rather nice to just sort of stand around or lean back with a gamepad while having some document filling up your screen. I implemented it as a half joke, but it turned out to be a killer feature.

In vim style, Zathura does just quit if you press q. I'm not necessarily the biggest fan of that either, but "oops, I pressed q" is something very different than "oops, I pressed back twice instead of once."